### ISSUE TITLE
Sklearn compatible multi feature transformer 

### ISSUE THREAD

**Issue by lcrmorin** [guillermo-navas-palencia/optbinning#343]
I have found myself in a position to need a practical way to apply binning + WoE encoding on all features of a dataset. As I was not able to find the solution within the package I have implemented a custom sklearn compatible version myself. Typically it works as follows:

```
OBT = OptBinningTransformer()
X_WoE = OBT.fit_transform(X, y)
```

I have made some effort to comment / sanitise / make it customisable / handle types / define some tests. Would it be a good idea to integrate this ? Where ?

**Comment by bmreiniger** [comment]
Doesn't `BinningProcess` do this, since the default transform method for a binary classification is WoE?

**Comment by lcrmorin** [comment]
It seems to be working after inputing column names... I somehow confused it with something else. 
Still a bit weird to have to specify column names.

**Comment by bmreiniger** [comment]
Oh, yes, I would advocate for maybe defaulting the list of columns to `None` which then applies to all columns (using names if pandas and `x{i} for i in range(m)` else).

(I also have a thin wrapper in my company's packaging for this sort of thing. I think I recall the priority of the transform metric specifications (in constructor and in transform) seemed backwards from what I'd expect too?)

**Comment by jnsofini** [comment]
I would advocate for setting the name to null or dummy values, such as C1, C2, etc., so that one can pass a numpy array in it.

**Comment by lcrmorin** [comment]
Revisiting this — the gap is smaller than a new class. BinningProcess.fit_transform(X, y) already does WoE-encoding across all features (DataFrame in → DataFrame out, columns preserved). Proposing three separable pieces instead:

1. variable_names optional. It's currently a required constructor arg — no way to say "just use all columns." Fix: default to None, resolve at fit() time (list(X.columns) for a DataFrame, [f"x{i}" ...] otherwise).

2. BinningProcess as sklearn.base.TransformerMixin. Tested this directly (patched a local copy, not just reasoning about it): adding the mixin doesn't touch the existing transform(X, metric=..., metric_special=..., ...) signature or fit_transform's sample_weight support — a class's own methods win over the mixin's defaults. Combined with get_feature_names_out() (already in flight as #382/#383), it also enables set_output(transform="pandas") — fit on a bare array, get a DataFrame with real column names back. Confirmed clone() and a full Pipeline(...).fit()/.predict() still work.

3. Transform-time params through a Pipeline. Today, metric=/metric_special= only work calling .transform() directly — Pipeline.transform(X) won't forward them to a middle step. Needs sklearn's metadata-routing API (set_transform_request()), a separate, bigger opt-in. Flagging for later, not proposing now.

Starting with 1 and 2 as focused PRs against BinningProcess directly. Shout if you'd rather see this as a new class instead.
